staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go

```diff
@@ -23,11 +23,23 @@ import (
 	"github.com/golang/glog"
 
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/audit"
 	"k8s.io/apiserver/pkg/authorization/authorizer"
 	"k8s.io/apiserver/pkg/endpoints/handlers/responsewriters"
 	"k8s.io/apiserver/pkg/endpoints/request"
 )
 
+const (
+	// Annotation key names set in advanced audit
+	decisionAnnotationKey = "authorization.k8s.io/decision"
+	reasonAnnotationKey   = "authorization.k8s.io/reason"
+	errorAnnotationKey    = "authorization.k8s.io/error"
```

Comment:
I don't think we should include errors here. Those should get printed in the debug logs. The reason could just be "error" in this case.

Reply:
makes sense (if reason is empty and error is non-nil, reason becomes "error")

Reply:
Done.